packages/utils/src/lib/git.integration.test.ts

```diff
@@ -60,4 +76,86 @@ describe('git utils', () => {
       'Backend/API/Startup.cs',
     );
   });
+
+  it('statusIsClean should return false if some changes are given', async () => {
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    await expect(statusIsClean(git)).resolves.toBe(false);
+  });
+
+  it('statusIsClean should return true if no changes are given', async () => {
+    await expect(statusIsClean(git)).resolves.toBe(true);
+  });
+
+  it('guardAgainstLocalChanges should throw if history is dirty', async () => {
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    await expect(guardAgainstLocalChanges(git)).rejects.toThrow(
+      'Working directory needs to be clean before we you can proceed. Commit your local changes or stash them.',
+    );
+  });
+
+  it('guardAgainstLocalChanges should not throw if history is clean', async () => {
+    await expect(guardAgainstLocalChanges(git)).resolves.toBeUndefined();
+  });
+
+  it('safeCheckout should checkout target branch in clean state', async () => {
+    await expect(git.branch()).resolves.toEqual(
+      expect.objectContaining({ current: 'master' }),
+    );
+    await expect(
+      safeCheckout('feature-branch', {}, git),
+    ).resolves.toBeUndefined();
+    await expect(git.branch()).resolves.toEqual(
+      expect.objectContaining({ current: 'feature-branch' }),
+    );
+  });
+
+  it('safeCheckout should throw if history is dirty', async () => {
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    await expect(safeCheckout('master', {}, git)).rejects.toThrow(
+      'Working directory needs to be clean before we you can proceed. Commit your local changes or stash them.',
+    );
+  });
+
+  it('safeCheckout should clean local changes and check out to master', async () => {
+    // needs to get reset to be clean
+    await mkdir(changesDir, { recursive: true });
+    await writeFile(join(changesDir, 'change.md'), '# hello-change\n');
+    // needs to get cleaned to be clean
+    await writeFile(join(baseDir, 'README.md'), '# hello-world-2\n');
```

Comment:
Could be valuable to add a check that these changes are not present anymore using `git.status()`.

Reply:
done